File: backend/src/main/java/com/romeo/VetLink/subscription/domain/Subscription.java
package com.romeo.VetLink.subscription.domain;

import com.romeo.VetLink.config.OwnedEntity;
import com.romeo.VetLink.shopItem.domain.ShopItem;
import com.romeo.VetLink.user.User;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.security.core.context.SecurityContextHolder;

import java.util.List;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "subscription")
public class Subscription extends OwnedEntity {
    @Id
    @GeneratedValue
    private Integer id;

    private String name;
    private String recurrence;
    private String price;
    private String shortDescription;
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
            name = "subscription_shop_item",
            joinColumns = @JoinColumn(name = "subscription_id"),
            inverseJoinColumns = @JoinColumn(name = "shop_item_id")
    )
    private List<ShopItem> shopItems;

    @Override
    public void prePersist(){
        User user = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        this.owner = user.getClinicId();
    }

    @Override
    public String toString() {
        return "Subscription{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", recurrence='" + recurrence + '\'' +
                ", price='" + price + '\'' +
                ", shortDescription='" + shortDescription + '\'' +
                '}';
    }

}

File: mobile/lib/model/subscription.dart


import 'package:vetlink/model/shop-item.dart';

class Subscription {
  final int id;
  final String name;
  final String recurrence;
  final String price;
  final String shortDescription;
  final List<ShopItem> shopItems;


  const Subscription({
    required this.id,
    required this.name,
    required this.recurrence,
    required this.price,
    required this.shortDescription,
    required this.shopItems,
  });

  static Subscription fromMap(Map<String, dynamic> map) {
    return Subscription(
      id: map['id'],
      name: map['name'],
      recurrence: map['recurrence'],
      price: map['price'],
      shortDescription: map['shortDescription'],
      shopItems: List<ShopItem>.from(
          map['shopItemDTOs'].map((item) => ShopItem.fromMap(item))
      ),
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'recurrence': recurrence,
      'price': price,
      'shortDescription': shortDescription,
      'shopItems': shopItems.map((item) => item.toMap()).toList(),
    };
  }
}
